tests/receivers/oracledb/bundled_test.go

```diff
@@ -0,0 +1,81 @@
+// Copyright Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build integration
+
+package tests
+
+import (
+	"fmt"
+	"path"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/signalfx/splunk-otel-collector/tests/testutils"
+)
+
+func TestOracledbDockerObserver(t *testing.T) {
+	testutils.SkipIfNotContainerTest(t)
+	if runtime.GOOS == "darwin" {
+		t.Skip("unable to share sockets between mac and d4m vm: https://github.com/docker/for-mac/issues/483#issuecomment-758836836")
+	}
+	tc := testutils.NewTestcase(t)
+	defer tc.PrintLogsOnFailure()
+	defer tc.ShutdownOTLPReceiverSink()
+
+	_, stop := tc.Containers(
+		testutils.NewContainer().WithContext(
+			path.Join(".", "testdata", "server"),
+		).WithName("oracledb").WithExposedPorts("1521:1521").WillWaitForHealth(5 * time.Minute),
+	)
+
+	defer stop()
+	fmt.Println("Now starting Otel Collector")
+
+	_, shutdown := tc.SplunkOtelCollectorContainer(
+		"otlp_exporter.yaml",
+		func(c testutils.Collector) testutils.Collector {
+			cc := c.(*testutils.CollectorContainer)
+			cc.Container = cc.Container.WithBinds("/var/run/docker.sock:/var/run/docker.sock:ro")
+			cc.Container = cc.Container.WillWaitForLogs("Discovering for next")
+			cc.Container = cc.Container.WithUser(fmt.Sprintf("999:%d", testutils.GetDockerGID(t)))
+			return cc
+		},
+
+		func(c testutils.Collector) testutils.Collector {
+			return c.WithEnv(map[string]string{
+				// runner seems to be slow
+				"SPLUNK_DISCOVERY_DURATION": "30s",
+				// confirm that debug logging doesn't affect runtime
+				"SPLUNK_DISCOVERY_LOG_LEVEL": "debug",
+			}).WithArgs(
+				"--discovery",
+				"--set", "splunk.discovery.receivers.oracledb.config.username=otel",
+				"--set", "splunk.discovery.receivers.oracledb.config.password=password",
```

Comment:
Mind doing something like this and setting an env var w/ the password to pick up expansion coverage?

```suggestion
				"--set", "splunk.discovery.receivers.oracledb.config.password='${ORACLE_PASSWORD}'",
```

Reply:
Want me to do env vars for the other settings also?